### DISCUSSION TITLE
project level editor settings?

### DISCUSSION THREAD

**Discussion by danbjoseph** [pulsar-edit/.github#207] [Pulsar Support] [answered]
is there a way to change editor settings automatically at the project level? e.g. i have the main settings for Tab Type set to "hard" but i'm working on a project that needs it set to "soft" and i don't want to adjust it in the main settings every time i switch projects. i tried to find this in the docs and with a web search but missed it or wasn't using the right words to search.

**Comment by savetheclocktower** [comment, marked as answer]
There's an API for this that was added to Atom, but which didn't ever have a built-in interface. But you can install either [atomic-management](https://web.pulsar-edit.dev/packages/atomic-management) or [project-config](https://web.pulsar-edit.dev/packages/project-config) to pull this off.

For both packages, you'd create a file inside an `.atom` directory at the root of your project — either `config.cson` (for `atomic-management`) or `config.json` (for `project-config`). You can then define overrides using the same format as your base `config.cson` file. Assuming you choose `atomic-management`, it'd look like this:

```coffeescript
"*":
  editor:
    softTabs: true
```
